search/src/Components/upload/upload_stepper.js

```diff
@@ -299,6 +287,28 @@ class UploadStepper extends React.Component {
         }
     }
 
+    nextHandler() {
+        const step = stepsByType[this.props.upload_type][this.state.activeStep].type
+        const currentStageValid = (step) => {
+            switch (step) {
+                case "categories":
+                    return [this.state.categories_saved, this.handleUpdateCategories]
```

Comment:
Using lists for heterogeneous data is confusing. If you want to mix multiple pieces of information, use an object with meaningful keys.